front50-web/src/main/java/com/netflix/spinnaker/front50/controllers/PipelineController.java

```diff
@@ -355,18 +492,29 @@ private void validatePipeline(final Pipeline pipeline, Boolean staleCheck) {
       }
     }
 
-    checkForDuplicatePipeline(
-        pipeline.getApplication(), pipeline.getName().trim(), pipeline.getId());
+    ensureCronTriggersHaveIdentifier(pipeline);
+
+    // Ensure cron trigger ids are regenerated if needed
+    if (Strings.isNullOrEmpty(pipeline.getId())
+        || (boolean) pipeline.getAny().getOrDefault("regenerateCronTriggerIds", false)) {
+      // ensure that cron triggers are assigned a unique identifier for new pipelines
+      pipeline.getTriggers().stream()
+          .filter(it -> "cron".equals(it.getType()))
+          .forEach(it -> it.put("id", UUID.randomUUID().toString()));
+    }
 
     final ValidatorErrors errors = new ValidatorErrors();
-    pipelineValidators.forEach(it -> it.validate(pipeline, errors));
 
+    // Run stale pipeline definition check
     if (staleCheck
         && !Strings.isNullOrEmpty(pipeline.getId())
         && pipeline.getLastModified() != null) {
       checkForStalePipeline(pipeline, errors);
     }
 
+    // Run other pre-configured validators
+    pipelineValidators.forEach(it -> it.validate(pipeline, errors));
+
```

Comment:
Honestly a LOT of the above operations LIKE the cron ids could be moved to pipelineValidators & shrink a lot of this code down :) and simplify it! Future enhancement maybe...